scripts/actions/utils/vendor-request.js

```diff
@@ -0,0 +1,91 @@
+const fetch = require('node-fetch');
+
+const makeRequest = async (url, options) => {
+  try {
+    const resp = await fetch(url.href, options);
+    const { response } = await resp.json();
+    const { code, data } = response;
+
+    if (code !== 'SUCCESS') {
+      console.dir(response, { depth: null });
+      throw new Error(code);
+    }
+
+    return data;
+  } catch (error) {
+    console.error(
+      `Unable to make a ${options.method} request to ${url.href}. (${error})`
+    );
+    process.exit(1);
```

Comment:
Rather than exiting, we should let the script calling this function deal with how to handle a failure. This may have adverse effects on the script since a lot of the requests happen concurrently.

Reply:
I've refactored the code to only call `process.exit` from the main entrypoint function.

We still log issues, but it's up to the caller to handle and thrown errors. At the moment, we're just catching them at the `main` function, but we can add some additional error response (i.e. communicating the errors via slack).

Reply:
Oh...just realized that we're still blowing away the queue, regardless if the file uploaded successfully or not. That's going to require a bit of work to implement. Gonna jump on that now.

Reply:
Resolved in https://github.com/newrelic/docs-website/pull/581/commits/f32c2b71774b6e4c30a07e393f38441d04c1b710